### PR TITLE
PER-4819 adding generic recommendation formatter

### PR DESCRIPTION
**Pull request by sqafabiano**
Adding transformer for adapting recommendations into the standard format.

### CODE REVIEW

File: tidal_per_transformers/transformers/generic_recommendation_formatter.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        self.module_name = module_name
+        self.module_version = module_version
+
+    def _transform(self, dataset: DataFrame) -> DataFrame:
+        return (dataset
+                .withColumn(c.TYPE, F.lit(self.item_type))
+                .withColumn(c.MODULE_ID, F.lit(f"{self.module_name}_v{self.module_version}"))
+                .withColumn(c.ID, F.explode(self.recommendations_col))
+                .groupby([F.col(self.hash_key).alias(c.HASH_KEY), c.MODULE_ID])
+                .agg(F.collect_list(F.struct(F.col(c.TYPE), F.col(c.ID))).alias(self.recommendations_col))
```
Comment (l-moamen) [inline tidal_per_transformers/transformers/generic_recommendation_formatter.py:31]: maybe we can add an over window by the hashkey as a partition, and then over the pos to insure the correct order

Comment (sqafabiano) [reply tidal_per_transformers/transformers/generic_recommendation_formatter.py:31]: Ok, you convinced me, better be extra safe :)

Comment (sqafabiano) [reply tidal_per_transformers/transformers/generic_recommendation_formatter.py:31]: I'm not being able to do this with a simple solution (I might be missing something though). Adding a window here makes the aggregation complain that the sorting key isn't part of the group by.

Comment (sqafabiano) [reply tidal_per_transformers/transformers/generic_recommendation_formatter.py:31]: That will give: 
`[[MISSING_AGGREGATION](https://docs.databricks.com/error-messages/error-classes.html#missing_aggregation)] The non-aggregating expression "pos" is based on columns which are not participating in the GROUP BY clause.
Add the columns or the expression to the GROUP BY, aggregate the expression, or use "any_value(pos)" if you do not care which of the values within a group is returned. SQLSTATE: 42803`

Comment (sqafabiano) [reply tidal_per_transformers/transformers/generic_recommendation_formatter.py]: Just to unblock myself. Updated the code with the non-straight solution (which is the same used by CollectRankedList)

**Review by l-moamen — APPROVED** [review]

**Comment by l-moamen** [comment]
Merging to use in my PR!

**Pull request merged**